### PR TITLE
Advanced theming

### PR DESCRIPTION
**Pull request by orangewolf**
this is a copy of #1727 to see if I can get github to load the diff. 

**Adds theming work from Palni/Palci**

This PR adds 3 custom home page themes, 2 custom show page themes and 2 search results themes. There is a new tab on the admin dashboard (settings/appearance) which allows admins to select themes.

Changes proposed in this pull request:
* Adds theme views
* Adds theme wireframes
* Adds theme css and js
* Adds methods to render the theme partials if a theme is selected
* Adds new tab on Appearance panel for themes with a theme selector
* Adds new content block for home page text
* Adds feature flags for Featured Researcher, Recently uploaded, Featured Works, Share your work button

## Screenshots
**Cultural Repository Home**
<details>
https://share.getcloudapp.com/xQu6jN1X
</details>

**Institutional Repository Home**
<details>
https://share.getcloudapp.com/o0uP7K10
</details>

**Neutral Repository Home**
<details>
https://share.getcloudapp.com/X6ulbnvN
</details>

**Cultural Show**
<details>
https://share.getcloudapp.com/E0ujKJXO
</details>

**Scholarly Show**
<details>
https://share.getcloudapp.com/xQu6jQb8
</details>

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -78,6 +78,18 @@ RSpec/InstanceVariable:
 RSpec/NamedSubject:
   Enabled: false
 
+RSpec/LetSetup:
+  Exclude:
```
Comment (orangewolf) [inline .rubocop.yml:82]: I'm not sure why we don't just fix these, but no matter what, we should not override them here.  Use the #rubocop:disable syntax at point of violation instead please.

File: Gemfile [hunk 2]
```diff
@@ -116,13 +122,11 @@ group :aws do
...
-
-gem 'codemirror-rails'
-gem 'riiif', '~> 1.1'
-
 gem 'bootstrap-datepicker-rails'
+gem 'codemirror-rails'
 gem 'parser', '~> 2.5.3'
+gem 'riiif', '~> 1.1'
+gem 'secure_headers'
+gem "sentry-raven"
```
Comment (orangewolf) [inline Gemfile:130]: notch8 specific gem sneaking in here.

File: app/models/featured_work.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
```
Comment (orangewolf) [inline app/models/featured_work.rb:2]: missing override

File: spec/controllers/hyrax/homepage_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,176 @@
+# frozen_string_literal: true
+
+# Copied from Hyrax v2.9.0 to add home_text content block to the index method - Adding themes
+RSpec.describe Hyrax::HomepageController, type: :controller, clean: true do
+  routes { Hyrax::Engine.routes }
```
Comment (orangewolf) [inline spec/controllers/hyrax/homepage_controller_spec.rb:5]: this is global and should use a let instead:
```
let(:routes) { Hyrax::Engine.routes }
```

File: .rubocop.yml (hunk 1 above)
Comment (labradford) [reply .rubocop.yml:82]: my mistake, I think I copied these over from another app by accident

**Comment by labradford** [comment]
I've made the changes on the original PR because I do not have access rights to push code directly to Hyku https://github.com/samvera/hyku/pull/1727

I had the same problem with the page loading and I ended up using Firefox to view the diff.

**Comment by orangewolf** [comment]
this was finished and merged in another PR

**Pull request closed without merging**